### PR TITLE
Introduce Whatsapp bot regex

### PR DESCRIPTION
**Pull request by benabbon**
User-agent format : WhatsApp/2.17.70 W

### CODE REVIEW

File: regexes.yaml [hunk 1]
```diff
@@ -27,11 +27,15 @@ user_agent_parsers:
   # Twitter
   - regex: '(Twitterbot)/(\d+)\.(\d+)'
     family_replacement: 'TwitterBot'
+
```
Comment (commenthol) [inline regexes.yaml]: This one in fact is the matching regex. Do we really need to name family "WhatsAppBot" ? IMO is better to update the regex in line 48

File: regexes.yaml [hunk 2]
```diff
@@ -27,11 +27,15 @@ user_agent_parsers:
...
+    
+  # WhatsApp
+  - regex: '(WhatsApp)/(\d+)\.(\d+).(\d+)'
+    family_replacement: 'WhatsAppBot'
 
   # Bots Pattern '/name-0.0'
   - regex: '/((?:Ant-)?Nutch|[A-z]+[Bb]ot|[A-z]+[Ss]pider|Axtaris|fetchurl|Isara|ShopSalad|Tailsweep)[ \-](\d+)(?:\.(\d+)(?:\.(\d+))?)?'
   # Bots Pattern 'name/0.0'
-  - regex: '\b(008|Altresium|Argus|BaiduMobaider|BoardReader|DNSGroup|DataparkSearch|EDI|Goodzer|Grub|INGRID|Infohelfer|LinkedInBot|LOOQ|Nutch|PathDefender|Peew|PostPost|Steeler|Twitterbot|VSE|WebCrunch|WebZIP|Y!J-BR[A-Z]|YahooSeeker|envolk|sproose|wminer)/(\d+)(?:\.(\d+)(?:\.(\d+))?)?'
+  - regex: '\b(008|Altresium|Argus|BaiduMobaider|BoardReader|DNSGroup|DataparkSearch|EDI|Goodzer|Grub|INGRID|Infohelfer|LinkedInBot|LOOQ|Nutch|PathDefender|Peew|PostPost|Steeler|Twitterbot|WhatsAppBot|VSE|WebCrunch|WebZIP|Y!J-BR[A-Z]|YahooSeeker|envolk|sproose|wminer)/(\d+)(?:\.(\d+)(?:\.(\d+))?)?'
```
Comment (commenthol) [inline regexes.yaml]: We have no UA-String matching "WhatsAppBot". Please remove.

File: regexes.yaml [hunk 3]
```diff
@@ -41,7 +45,7 @@ user_agent_parsers:
...
...|ichiro(?:/mobile)?|IconSurf|IlTrovatore(?:-Setaccio)?|InfuzApp|Innovazion Crawler|InternetArchive|IP2[a-z]+Bot|jbot\b|KaloogaBot|Kraken|Kurzor|larbin|LEIA|LesnikBot|Linguee Bot|LinkAider|LinkedInBot|Lite Bot|Llaut|lycos|Mail\.RU_Bot|masidani_bot|Mediapartners-Google|Microsoft .*? Bot|mogimogi|mozDex|MJ12bot|msnbot(?:-media *)?|msrbot|netresearch|Netvibes|NewsGator[^/]*|^NING|Nutch[^/]*|Nymesis|ObjectsSearch|Orbiter|OOZBOT|PagePeeker|PagesInventory|PaxleFramework|Peeplo Screenshot Bot|PlantyNet_WebRobot|Pompos|Read%20Later|Reaper|RedCarpet|Retreiver|Riddler|Rival IQ|scooter|Scrapy|Scrubby|searchsight|seekbot|semanticdiscovery|Simpy|SimplePie|SEOstats|SimpleRSS|SiteCon|Slackbot-LinkExpanding|Slack-ImgProxy|Slurp|snappy|Speedy Spider|Squrl Java|TheUsefulbot|ThumbShotsBot|Thumbshots\.ru|TwitterBot|WhatsappBot|URL2PNG|Vagabondo|VoilaBot|^vortex|Votay bot|^voyager|WASALive.Bot|Web-sniffer|WebThumb|WeSEE:[A-z]+|WhatWeb|WIRE|WordPress|Wotbox|www\.almaden\.ibm\.com|Xenu(?:.s)? Link Sleuth|Xerka [A-z]+Bot|yacy(?:bot)?|Yahoo[a-z]*Seeker|Yahoo! Slurp|Yandex\w+|YodaoBot(?:-[A-z]+)?|YottaaMonitor|Yowedo|^Zao|^Zao-Crawler|ZeBot_www\.ze\.bz|ZooShot|ZyBorg)(?:[ /]v?(\d+)(?:\.(\d+)(?:\.(\d+))?)?)?'
```
Comment (commenthol) [inline regexes.yaml]: Please change "WhatsappBot" to "WhatsApp". That should do the match.

File: tests/test_ua.yaml [hunk 4]
```diff
@@ -275,6 +275,12 @@ test_cases:
     major: '1'
     minor: '0'
     patch:
+    
+  - user_agent_string: 'WhatsApp/2.17.70 W'
+    family: 'WhatsAppBot'
```
Comment (commenthol) [inline tests/test_ua.yaml]: Would need a renaming to "WhatsApp"

File: regexes.yaml (hunk 1 above)
Comment (benabbon) [inline regexes.yaml]: Updated

**Comment by commenthol** [comment]
Merged with corrections in #224. Thanks for your contribution.

**Pull request closed without merging**